src/v2/view-builder/views/ov/EnrollAuthenticatorOVPollView.js

```diff
@@ -0,0 +1,118 @@
+import { View, loc } from 'okta';
+import BaseForm from '../../internals/BaseForm';
+import { isAndroid, isIOS } from '../../../../util/BrowserFeatures';
+import BaseAuthenticatorView from '../../components/BaseAuthenticatorView';
+import AuthenticatorEnrollFooter from '../../components/AuthenticatorEnrollFooter';
+import Link from '../../components/Link';
+import { getSwitchOVEnrollChannelLink } from '../../utils/LinksUtil';
+import polling from '../shared/polling';
+import hbs from 'handlebars-inline-precompile';
+
+const Body = BaseForm.extend(Object.assign(
+  polling,
+  {
+    title () {
+      const selectedChannel = this.options.appState.get('currentAuthenticator').contextualData.selectedChannel;
+      let title;
+      switch (selectedChannel) {
+      case 'email':
+        title = loc('oie.enroll.okta_verify.setup.email.title', 'login');
+        break;
+      case 'sms':
+        title = loc('oie.enroll.okta_verify.setup.sms.title', 'login');
+        break;
+      default:
+        title = loc('oie.enroll.okta_verify.setup.title', 'login');
+      }
+      return title;
+    },
+    className: 'oie-enroll-ov-poll',
+    noButtonBar: true,
+    initialize () {
+      BaseForm.prototype.initialize.apply(this, arguments);
+      if ((isAndroid() || isIOS()) &
+        this.options.appState.get('currentAuthenticator').contextualData.selectedChannel === 'qrcode') {
+        this.options.appState.trigger('switchForm', 'enroll-with-another-channel');
+      }
+      this.listenTo(this.model, 'error', this.stopPolling);
+      this.startPolling();
+    },
+    getUISchema () {
+      const schema = [];
+      const contextualData = this.options.appState.get('currentAuthenticator').contextualData;
+      const selectedChannel = contextualData.selectedChannel;
+      if (selectedChannel === 'qrcode') {
+        schema.push({
+          View: View.extend({
+            template: hbs`
```

Comment:
we should just create one hbs template and try to reuse it for each channel type.

Reply:
or may be just one view that can return different channel type views based on options passed

Reply:
good point. I like moving this to one view and move some logic into that.  I am still expecting some UX changes and plan to work on them after my courage rotation. Will refactor based on UX changes. Might have to split into three views too. 

Jira for that https://oktainc.atlassian.net/browse/OKTA-320368